client/client.go

```diff
@@ -0,0 +1,130 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/eigenda"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const (
+	// NOTE: this will need to be updated as plasma's implementation changes
+	decodingOffset = 2
+)
+
+// TODO: Add support for custom http client option
+type Config struct {
+	URL string
+}
+
+// ProxyClient is an interface for communicating with the EigenDA proxy server
+type ProxyClient interface {
+	Health() error
+	GetData(ctx context.Context, cert *common.Certificate, domain common.DomainType) ([]byte, error)
+	SetData(ctx context.Context, b []byte) (*common.Certificate, error)
+}
+
+// client is the implementation of ProxyClient
+type client struct {
```

Comment:
* Why not expose the concrete type?
* Naming-wise, I think "eigenda" should either be in the package name or in the struct/interface name.

Reply:
Is this client specific to OP stack or general?

Reply:
general